### PR TITLE
Create tasks to generate CSV 

### PR DESCRIPTION
**Pull request by michellemtchai**
- format for the CSV generated:
```
CDF info:
  course code,emailaddress,studentnumber,familyname,givennames,studentdepartment,utorid

Offers:
  course code, course title, offer hours, student number, familyname, givennames, student status, student department, email address, round_id

Applicant transcript access:
  studentnumber, family name, givennames,  grant (y or n), email address
```
- Create some assignments on the ABC view before running the following commands
- The following are the commands to generate the CSV
  - `docker-compose run rails-app rake db:csv:cdf`
    - output is located at `db/seeds/cdf_info.csv`
  - `docker-compose run rails-app rake db:csv:offers`
    - output is located at `db/seeds/offers.csv`
  - `docker-compose run rails-app rake db:csv:transcript_access`
    - output is located at `db/seeds/transcript_access.csv`

- The following are the list of feedbacks you may get:
  - `Warning: You have not made any assignments yet. Operation aborted` when your have no assignments and ran either the `cdf` or `offers` exporter
  - `Success: CDF info CSV file created`, `Success: Offer CSV file created`, and `Suceess: Transcript Access CSV file created` are for when the file creation is successful

Assumptions:
- CDF info is only made for those applicant who are assigned
    - this means that if there are no assignments, there are no CDF info output
- transcript access outputs the access permission for all applicants in the system
- the services `CsvGenerator` can be connect to a controller to download the CSV file from the front-end

### CODE REVIEW

File: app/services/csv_generator.rb [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      ])
+      data = (data + csv_string)
+    end
+    return data
+  end
+
+  def generate_csv_string(row)
+    csv_string = ""
+    row.each do |item|
+      csv_string = (csv_string + item + ", ")
```
Comment (reidka) [inline app/services/csv_generator.rb]: I was wondering about the space.  Excel does open the file without complaining, but I think that this doesn't satisfy the csv format specs.  (I didn't take the time to go look them up, but we've been bitten by this in MarkUs before.)  We have two options: leave it as is, and leave a TODO note to fix it up, or use a csv library to create the csv file.  (There are other issues like space separated elements, and elements with commas in them which won't happen with this data)

**Review by reidka — COMMENTED** [review]
This seems to work as expected.  I added a note about the csv format.

File: app/services/csv_generator.rb (hunk 1 above)
Comment (michellemtchai) [reply app/services/csv_generator.rb]: I took a look at the specs for CSV on this [https://tools.ietf.org/html/rfc4180](page) and it does say that space are included as part of the field in rule 4, so I'll change line 92 from `...item+", ")` to `...item+",")`.

Comment (michellemtchai) [reply app/services/csv_generator.rb]: I changed the CSV to using a CSV generation library now.

**Review by jmzaleski — APPROVED** [review]
comment the functions to add the word "export" somewhere in the file..

**Pull request merged**